nibe/connection/encoders.py

```diff
@@ -38,46 +38,31 @@ class CoilDataEncoder:
     def __init__(self, word_swap: bool = True) -> None:
         self._word_swap = word_swap
 
-    def encode(self, coil: Coil):
-        value = coil.value
+    def encode(self, coil_data: CoilData) -> bytes:
         try:
-            assert value is not None, "Unable to encode None value"
-            if coil.has_mappings:
-                return self._pad(
-                    self._get_parser(coil), coil.get_reverse_mapping_for(value)
-                )
+            assert coil_data.is_valid, "Invalid coil data"
```

Comment:
Maybe we should change is_valid() to check() or verify() and have it pass along the exception. Then we retain the reason for the failure.

Reply:
Do you mean `verify()` should throw exception or just return it?

Reply:
Throw it was my intention.